cluster-autoscaler/cloudprovider/aws/CA_with_AWS_IAM_OIDC.md

```diff
@@ -0,0 +1,171 @@
+- In the "Add/Edit Auto Scaling Group Tags" window, please make sure you enter the following tags by replacing 'awsExampleClusterName' with the name of your EKS cluster. Then, choose "Save".
+
+| Plugin | README |
+| ------ | ------ |
+| Key: | k8s.io/cluster-autoscaler/enabled |
+| Key: | k8s.io/cluster-autoscaler/awsExampleClusterName |
+
+Note: The keys for the tags that you entered don't have values. Cluster Autoscaler ignores any value set for the keys.
+
+- Create an IAM Policy for cluster autoscaler and to enable AutoDiscovery. 
+
+```sh
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "autoscaling:DescribeAutoScalingGroups",
+                "autoscaling:DescribeAutoScalingInstances",
+                "autoscaling:DescribeLaunchConfigurations",
+                "autoscaling:DescribeTags",
+                "autoscaling:SetDesiredCapacity",
+                "autoscaling:TerminateInstanceInAutoScalingGroup"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+NOTE: autoscaling:DescribeTags is very important if you are making use of the AutoDiscovery feature of the Cluster AutoScaler. 
+
+- Attach the above created policy to the *instance role* that's attached to your Amazon EKS worker nodes.
+- Download a deployment example file provided by the Cluster Autoscaler project on GitHub, run the following command:
+
+```sh
+$ wget https://raw.githubusercontent.com/kubernetes/autoscaler/master/cluster-autoscaler/cloudprovider/aws/examples/cluster-autoscaler-autodiscover.yaml
+```
+
+- Open the downloaded YAML file in an editor. 
+
+##### Change 1: 
+
+Set the EKS cluster name (awsExampleClusterName) and environment variable (us-east-1) based on the following example. 
+
+```sh
+    spec:
+      serviceAccountName: cluster-autoscaler
+      containers:
+        - image: gcr.io/google-containers/cluster-autoscaler:v1.14.6     #cluster-autoscaler image
+          name: cluster-autoscaler
+          resources:
+            limits:
+              cpu: 100m
+              memory: 300Mi
+            requests:
+              cpu: 100m
+              memory: 300Mi
+          command:
+            - ./cluster-autoscaler
+            - --v=4
+            - --stderrthreshold=info
+            - --cloud-provider=aws
+            - --skip-nodes-with-local-storage=false
+            - --expander=least-waste
+            - --node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/<<awsExampleClusterName>>
+          env:
+            - name: AWS_REGION
+              value: <<us-east-1>>
+```
+
+##### Change 2: 
+
+To use IAM with OIDC, you will have to make the below changes to the file as well. 
+
+```sh
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    k8s-addon: cluster-autoscaler.addons.k8s.io
```

Comment:
Where does label come from?

Reply:
Refer commit: [5dffd75](https://github.com/kubernetes/autoscaler/commit/5dffd75f5bc9b8878d22f9cea07539e6668a98b2) 

The addons resizer is used to extend the functionality of the cluster and is published in AWS repositories.